resources/[soz]/soz-core/src/client/shop/tattoo.shop.provider.ts

```diff
@@ -0,0 +1,239 @@
+import { ShopBrand, ShopsConfig, ShopTattooConfig } from '@public/config/shops';
+import { ShopTattooProducts } from '@public/config/tattoos';
+import { OnNuiEvent } from '@public/core/decorators/event';
+import { Inject } from '@public/core/decorators/injectable';
+import { Provider } from '@public/core/decorators/provider';
+import { Tick, TickInterval } from '@public/core/decorators/tick';
+import { wait } from '@public/core/utils';
+import { NuiEvent, ServerEvent } from '@public/shared/event';
+import { MenuType } from '@public/shared/nui/menu';
+import { Skin } from '@public/shared/player';
+import { Vector3 } from '@public/shared/polyzone/vector';
+import { Ok } from '@public/shared/result';
+import { TattooShopItem } from '@public/shared/shop';
+
+import { AnimationService } from '../animation/animation.service';
+import { CameraService } from '../camera';
+import { Notifier } from '../notifier';
+import { InputService } from '../nui/input.service';
+import { NuiMenu } from '../nui/nui.menu';
+import { PlayerService } from '../player/player.service';
+
+type ShopTattooState = {
+    camera: number;
+    cameraCurrentVariation: number;
+    currentSelectedCategory: string;
+    scaleform: number;
+};
+
+@Provider()
+export class TattooShopProvider {
+    @Inject(NuiMenu)
+    private nuiMenu: NuiMenu;
+
+    @Inject(PlayerService)
+    private playerService: PlayerService;
+
+    @Inject(CameraService)
+    private cameraService: CameraService;
+
+    @Inject(InputService)
+    private inputService: InputService;
+
+    @Inject(Notifier)
+    private notifier: Notifier;
+
+    @Inject(AnimationService)
+    private animationService: AnimationService;
+
+    private state: ShopTattooState = null;
+
+    public async openShop(brand: ShopBrand, shop: string) {
+        const categories = ShopTattooConfig;
+        const ped = PlayerPedId();
+        const overlayField = GetEntityModel(ped) === GetHashKey('mp_m_freemode_01') ? 'HashNameMale' : 'HashNameFemale';
```

Comment:
As I remember, for performance, you should use joaat function here instead of GetHashKey

Reply:
Replacer by `PlayerPedHash.Male`